Lib/test/test_typing.py

```diff
@@ -1954,6 +1968,14 @@ def test_var_substitution(self):
         self.assertEqual(C5[int, str, float],
                          Callable[[typing.List[int], tuple[str, int], float], int])
 
+    def test_type_subst_error(self):
+        Callable = self.Callable
+        P = ParamSpec('P')
+        T = TypeVar('T')
+
+        with self.assertRaises(TypeError):
+            Callable[P, T][0, int]
```

Comment:
The error message is unusually good here, so let's assert on it:

```suggestion
        pat = "Expected a list of types, an ellipsis, ParamSpec, or Concatenate."

        with self.assertRaisesRegex(TypeError, pat):
            Callable[P, T][0, int]
```